rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/EvaluatorUtil.java

```diff
@@ -118,6 +112,28 @@ public static <T> InterruptibleFuture<T> runEvaluator(String task, CompletableFu
         });
     }
 
+    private static String formatMessage(Throw e) {
+        String message = "";
+        IValue thrown = e.getException();
+        if (thrown instanceof IString) {
+            message = ((IString)thrown).getValue();
+        }
+        else {
+            LimitedResultWriter res = new LimitedResultWriter(512);
+            try {
+                new StandardTextWriter(false).write(thrown, res);
+            }
+            catch (/*IO Limit*/ RuntimeException | IOException _ignored) {
+            }
+            message = res.toString();
+        }
+        return message + "\nat: " + e.getLocation();
+    }
+
+    private static String formatMessage(StaticError e) {
+        return "Static error in rascal code: " + e.getMessage() + "\nat: " + e.getLocation();
```

Comment:
Don't have to mention the programming language. Also isn't the location already reflected in the metadata? Probably we can elide the location here because it is accessible via the UI.

Reply:
Maybe true for the Static Errors indeed.